api/processors/group.py

```diff
@@ -0,0 +1,25 @@
+from api.models import Group
+from api.models.choices.statuses import (
+    CoordinatorStatus,
+    GroupStatus,
+    StudentStatus,
+    TeacherStatus,
+)
+from api.processors.base import Processor
+
+
+class GroupProcessor(Processor):
+    @classmethod
+    def start(cls, group: Group) -> None:
+        cls._set_status(obj=group, status=GroupStatus.WORKING)
+
+        # FIXME things below depend on teacher's/coordinator's params, just showing general logic
+
+        for coordinator in group.coordinators.iterator():
+            cls._set_status(obj=coordinator, status=CoordinatorStatus.WORKING_OK)
```

Comment:
Here and below: Now you do N hits to db to update N instances. This could be done by one `update` query for each related model

Reply:
Fixed